src/web/components/SignInGate/SignInGateSelector.tsx

```diff
@@ -208,19 +220,25 @@ export const SignInGateSelector = ({
 	const gateVariant: SignInGateComponent | null =
 		testVariantToGateMapping?.[currentTest.variant];
 
+	if (gateVariant) {
+		// eslint-disable-next-line @typescript-eslint/no-floating-promises
+		gateVariant
+			.canShow(CAPI, !!isSignedIn, currentTest)
+			.then(setCanShowGate);
+	}
+
 	return (
 		<>
 			{/* Sign In Gate Display Logic */}
-			{showGate &&
-				gateVariant?.canShow(CAPI, !!isSignedIn, currentTest) && (
-					<ShowSignInGate
-						CAPI={CAPI}
-						abTest={currentTest}
-						setShowGate={setShowGate}
-						signInUrl={signInUrl}
-						gateVariant={gateVariant}
-					/>
-				)}
+			{!isGateDismissed && canShowGate ? (
+				<ShowSignInGate
+					CAPI={CAPI}
+					abTest={currentTest}
+					setShowGate={(show) => setIsGateDismissed(!show)}
+					signInUrl={signInUrl}
+					gateVariant={gateVariant}
+				/>
+			) : null}
```

Comment:
```suggestion
			{!isGateDismissed && canShowGate &&
				<ShowSignInGate
					CAPI={CAPI}
					abTest={currentTest}
					setShowGate={(show) => setIsGateDismissed(!show)}
					signInUrl={signInUrl}
					gateVariant={gateVariant}
				/>
			}
```

should achieve the same thing without having to rely on the ternary operator